task_processing/plugins/kubernetes/task_config.py

```diff
@@ -104,6 +107,18 @@ def _valid_volumes(volumes: Sequence["DockerVolume"]) -> Tuple[bool, Optional[st
     return (True, None)
 
 
+def _valid_secret_envs(secret_envs: Mapping[str, "SecretEnvSource"]) -> Tuple[bool, Optional[str]]:
+    # XXX: Note we are not validating existence of secret in k8s here, leave that to creation of pod
```

Comment:
unless we want to want to hit the k8s api here, probably safe to remove the `XXX` TODO marker :)
```suggestion
    # Note we are not validating existence of secret in k8s here, leave that to creation of pod
```